syntax = "proto3";

import "utilities.proto";
import "types.proto";

package osirixgrpc;

// Message responses
message BrowserControllerDatabaseSelectionResponse{
	Status status = 1;
	repeated DicomSeries series = 2;
	repeated DicomStudy studies = 3;
}

message BrowserControllerOpenViewerFromImagesResponse{
	Status status = 1;
	ViewerController viewer = 2;
}

// Message requests
message BrowserControllerCopyFilesIfNeededRequest{
	BrowserController browser = 1;
	repeated string paths = 2;
}

message BrowserControllerOpenViewerFromImagesRequest{
	message FrameImages{
		repeated DicomImage images = 1;
	}
	BrowserController browser = 1;
	repeated FrameImages frames = 2;
	bool movie = 3;
}